File: Quiz.Entity/Models/Subject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Quiz.Entity.Models
{
    public class Subject
    {
        [Key]
        public int SubjectId { get; set; }
        public int ProgramId { get; set; }
        public string Name { get; set; }
        public string Part { get; set; }
        public string Code { get; set; }
        public bool IsActive { get; set; }
        public int SerialNo { get; set; }
    }
}

File: Quiz.Android/lib/models/subject.dart
class Subject {
  final int subjectId;
  final String name;
  final String? part;
  final String code;
  final bool isActive;
  final int programId;

  const Subject({
    required this.subjectId,
    required this.name,
    this.part,
    required this.code,
    required this.isActive,
    required this.programId,
  });

  factory Subject.fromJson(Map<String, dynamic> json) {
    return Subject(
      subjectId: json['subjectId'],
      name: json['name'],
      part: json['part'],
      code: json['code'],
      isActive: json['isActive'],
      programId: json['programId'],
    );
  }
}
